### ISSUE TITLE
[bug] Don't know how to make <C:\...\Hostx64\vcvarsall.bat.>

### ISSUE THREAD

**Issue by Siech0** [conan-io/conan#6159]
I am having an issue building boost_system/1.69.0@bincrafters/stable. From the logs it seems that it is searching for vcvarsall.bat in an incorrect folder: <_.../VC/Tools/MSVC/VERSION/bin/Hostx64/vcvarsall.bat>_, however, the actual location of the file in my (fresh) installation of visual studio is _<.../VC/Auxiliary/Build>_.

### Environment Details 
  * Operating System + version: Windows 10 Professional 64-bit (10.0, Build 18362)
  * Compiler+version: Visual Studio 2019 Community 16.3.10
  * Conan version: 1.20.4
  * Python version: 3.8.0

_(This environment is a completely fresh installation of windows 10 + entirely new installations of required software)_

### Steps to reproduce (Include if Applicable)
conanfile.txt:
```
[requires]
boost_system/1.69.0@bincrafters/stable

[generators]
cmake_find_package

```
project structure:
```
project\
     conanfile.txt
     CMakeLists.txt
     build\   
```

install command:
`[from build\] conan install .. --build missing`

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
D:\dev\c++\rest\build>conan install .. --build missing
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

boost_system/1.69.0@bincrafters/stable: WARN: Package is corrupted, removing folder: C:\.conan\d8b70d\1
conanfile.txt: Installing package
Requirements
    boost_config/1.69.0@bincrafters/stable from 'bincrafters' - Cache
    boost_predef/1.69.0@bincrafters/stable from 'bincrafters' - Cache
    boost_system/1.69.0@bincrafters/stable from 'bincrafters' - Cache
    boost_winapi/1.69.0@bincrafters/stable from 'bincrafters' - Cache
Python requires
    boost_base/1.69.0@bincrafters/stable
Packages
    boost_config/1.69.0@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    boost_predef/1.69.0@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    boost_system/1.69.0@bincrafters/stable:1cfc588b26e59ddc9e1094237b6c5153e7f2f0bc - Build
    boost_winapi/1.69.0@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
Build requirements
    boost_build/1.69.0@bincrafters/stable from 'bincrafters' - Cache
    boost_generator/1.69.0@bincrafters/stable from 'bincrafters' - Cache
Build requirements packages
    boost_build/1.69.0@bincrafters/stable:456f15897172eef340fcbac8a70811f2beb26a93 - Cache
    boost_generator/1.69.0@bincrafters/stable:6ec6439c54a97da1cfddff786d1ea15988300369 - Cache

boost_build/1.69.0@bincrafters/stable: Already installed!
boost_config/1.69.0@bincrafters/stable: Already installed!
boost_predef/1.69.0@bincrafters/stable: Already installed!
boost_generator/1.69.0@bincrafters/stable: Already installed!
boost_winapi/1.69.0@bincrafters/stable: Already installed!
boost_system/1.69.0@bincrafters/stable: Applying build-requirement: boost_generator/1.69.0@bincrafters/stable
boost_system/1.69.0@bincrafters/stable: Applying build-requirement: boost_build/1.69.0@bincrafters/stable
boost_system/1.69.0@bincrafters/stable: WARN: Build folder is dirty, removing it: C:\.conan\0d1287\1
boost_system/1.69.0@bincrafters/stable: Copying sources to build folder
boost_system/1.69.0@bincrafters/stable: Building your package in C:\.conan\0d1287\1
boost_system/1.69.0@bincrafters/stable: Generator boost created jamroot
boost_system/1.69.0@bincrafters/stable: Generator boost created boostcpp.jam
boost_system/1.69.0@bincrafters/stable: Generator boost created project-config.jam
boost_system/1.69.0@bincrafters/stable: Generator boost created short_path.cmd
boost_system/1.69.0@bincrafters/stable: Calling build()
boost_system/1.69.0@bincrafters/stable: C:\.conan\0d1287\1: b2 -j8 -d+1 -a --hash=yes --debug-configuration --layout=system    system-build
notice: found boost-build.jam at C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/boost-build.jam
notice: loading Boost.Build from C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/kernel
C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util\os.jam:50: Unescaped special character in argument ([^.]*)$
C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util\os.jam:204: Unescaped special character in argument os:
C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util\os.jam:205: Unescaped special character in argument os:
notice: Searching 'C:\Users\gamer' 'C:\Users\gamer' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src/kernel' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src\kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/build' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/tools' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/contrib' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/.' for site-config configuration file 'site-config.jam'.
notice: Configuration file 'site-config.jam' not found in 'C:\Users\gamer' 'C:\Users\gamer' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src/kernel' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src\kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/build' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/tools' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/contrib' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/.'.
notice: Searching 'C:\Users\gamer' 'C:\Users\gamer' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src/kernel' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src\kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/build' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/tools' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/contrib' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/.' for user-config configuration file 'user-config.jam'.
notice: Configuration file 'user-config.jam' not found in 'C:\Users\gamer' 'C:\Users\gamer' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src/kernel' 'C:\Users\gamer\.conan\data\boost_build\1.69.0\bincrafters\stable\package\456f15897172eef340fcbac8a70811f2beb26a93\share\boost-build\src\kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/kernel' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/util' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/build' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/tools' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/contrib' 'C:/Users/gamer/.conan/data/boost_build/1.69.0/bincrafters/stable/package/456f15897172eef340fcbac8a70811f2beb26a93/share/boost-build/src/.'.
notice: Searching '.' for project-config configuration file 'project-config.jam'.
notice: Loading project-config configuration file 'project-config.jam' from '.'.
notice: will use 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/x64/cl.exe' for msvc, condition <toolset>msvc-16.0
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>/<address-model>32', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>x86/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>x86/<address-model>32', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>/<address-model>64', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>x86/<address-model>64', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>ia64/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>ia64/<address-model>64', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>arm/<address-model>', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
notice: [msvc-cfg] condition: '<toolset>msvc-16.0/<architecture>arm/<address-model>32', setup: 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/vcvarsall.bat'
===============MPI Auto-configuration===============
Found Microsoft MPI: C:\Program Files\Microsoft MPI
Found Microsoft MPI SDK: C:\Program Files (x86)\Microsoft SDKs\MPI
MPI build features:
<include>/C:/Program Files (x86)/Microsoft SDKs/MPI/Include <address-model>64:<library-path>/C:/Program Files (x86)/Microsoft SDKs/MPI/Lib/x64 <library-path>/C:/Program Files (x86)/Microsoft SDKs/MPI/Lib/x86 <find-static-library>msmpi <toolset>msvc:<define>_SECURE_SCL=0
MPI launcher: "C:\Program Files\Microsoft MPI\Bin\mpiexec.exe" -n
====================================================
don't know how to make <p/C:/.conan/0d1287/1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\bin\Hostx64\vcvarsall.bat
...found 120 targets...
...updating 15 targets...
...can't find 1 target...
...can't make 2 targets...
...skipped <pbin\standalone\msvc\ab605d3a191801849fe1e1e55b908353>msvc-setup.read for lack of <p/C:/.conan/0d1287/1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\bin\Hostx64\vcvarsall.bat...
...skipped <pbin\standalone\msvc\ab605d3a191801849fe1e1e55b908353>msvc-setup.bat for lack of <p/C:/.conan/0d1287/1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\bin\Hostx64\vcvarsall.bat...
...skipped <pbin\system\build\aca5cb0fcccc3d2990e4e006ef019df6>error_code.obj for lack of <pbin\standalone\msvc\ab605d3a191801849fe1e1e55b908353>msvc-setup.nup...
...skipped <pbin\system\build\aca5cb0fcccc3d2990e4e006ef019df6>libboost_system.lib for lack of <pbin\system\build\aca5cb0fcccc3d2990e4e006ef019df6>error_code.obj...
...skipped <psystem\lib>libboost_system.lib for lack of <pbin\system\build\aca5cb0fcccc3d2990e4e006ef019df6>libboost_system.lib...
common.copy system\lib\jamroot.jam
bin\d41d8cd98f00b204e9800998ecf8427e\system-lib\jamroot.jam
        1 file(s) copied.
...skipped 5 targets...
...updated 12 targets...
boost_system/1.69.0@bincrafters/stable:
boost_system/1.69.0@bincrafters/stable: ERROR: Package '1cfc588b26e59ddc9e1094237b6c5153e7f2f0bc' build failed
boost_system/1.69.0@bincrafters/stable: WARN: Build folder C:\.conan\0d1287\1
ERROR: boost_system/1.69.0@bincrafters/stable: Error in build() method, line 147
        self._build_common()
while calling '_build_common', line 170
        self.run(" ".join(b2_command))
        ConanException: Error 1 while executing b2 -j8 -d+1 -a --hash=yes --debug-configuration --layout=system    system-build
```

**Comment by czoido** [comment]
Hi @Siech0, 
I think that this has not to do with Conan but that is an issue for boost 1.69 using VS2019 due to the inability of boost-build 1.69.0 to use/discover VS2019 that was fixed for the 1.70 release.
There are more people that have experienced this issue for this recipe:
https://github.com/bincrafters/community/issues/781
I only could suggest using the whole boost recipe from conan-center.

**Comment by memsharded** [comment]
Closing as stale/not an issue.
